diff --git a/src/passo/AlternateMinimization.C b/src/passo/AlternateMinimization.C
index 67a10c1..8ef3730 100644
--- a/src/passo/AlternateMinimization.C
+++ b/src/passo/AlternateMinimization.C
@@ -322,7 +322,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_AlternateMinimization_Solver(SNES s
     snes->ops->reset          = SNESReset_passo_AlternateMinimization_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/ContactStabilizedNewmark.C b/src/passo/ContactStabilizedNewmark.C
index 767b2b1..7ca3c74 100644
--- a/src/passo/ContactStabilizedNewmark.C
+++ b/src/passo/ContactStabilizedNewmark.C
@@ -339,7 +339,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_ContactStabilizedNewmark(SNES snes)
     snes->ops->reset          = SNESReset_ContactStabilizedNewmark;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     //snes->pcside = PC_LEFT;
     
diff --git a/src/passo/FAS.C b/src/passo/FAS.C
index 6961935..191f234 100644
--- a/src/passo/FAS.C
+++ b/src/passo/FAS.C
@@ -255,7 +255,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_FAS_Solver(SNES snes)
     snes->ops->reset          = SNESReset_passo_FAS_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/RMTR.C b/src/passo/RMTR.C
index 0d016b5..883be68 100644
--- a/src/passo/RMTR.C
+++ b/src/passo/RMTR.C
@@ -252,7 +252,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_RMTR_Solver(SNES snes)
     snes->ops->reset          = SNESReset_passo_RMTR_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/SemismoothNewton.C b/src/passo/SemismoothNewton.C
index 134678c..b7df405 100644
--- a/src/passo/SemismoothNewton.C
+++ b/src/passo/SemismoothNewton.C
@@ -336,7 +336,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_SemismoothNewton(SNES snes)
     snes->ops->reset          = SNESReset_SemismoothNewton;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     //snes->pcside = PC_LEFT;
     
diff --git a/src/passo/SemismoothNewtonEqualityConst.C b/src/passo/SemismoothNewtonEqualityConst.C
index 3a6ca17..2d50b6d 100644
--- a/src/passo/SemismoothNewtonEqualityConst.C
+++ b/src/passo/SemismoothNewtonEqualityConst.C
@@ -393,7 +393,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_SemismoothNewtonEqualityConst(SNES snes)
     snes->ops->reset          = SNESReset_SemismoothNewtonEqualityConst;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     //snes->pcside = PC_LEFT;
     
diff --git a/src/passo/benzi.C b/src/passo/benzi.C
index 3d09e01..40b57e7 100644
--- a/src/passo/benzi.C
+++ b/src/passo/benzi.C
@@ -186,9 +186,9 @@ SNESCreate_passo_Benzi_Solver(SNES snes)
   snes->ops->reset = SNESReset_passo_Benzi_Solver;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
-  // snes->pcside = PC_LEFT;
+  // snes->npcside = PC_LEFT;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
diff --git a/src/passo/contact.C b/src/passo/contact.C
index e193e1d..4b90276 100644
--- a/src/passo/contact.C
+++ b/src/passo/contact.C
@@ -446,9 +446,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_contact(SNES snes)
     snes->ops->reset          = SNESReset_passo_contact;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
-    //snes->pcside = PC_LEFT;
+    //snes->npcside = PC_LEFT;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/gaspin.C b/src/passo/gaspin.C
index ec8566f..1343745 100644
--- a/src/passo/gaspin.C
+++ b/src/passo/gaspin.C
@@ -224,9 +224,9 @@ SNESCreate_passo_GASPIN_Solver(SNES snes)
   snes->ops->reset = SNESReset_passo_GASPIN_Solver;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
-  snes->pcside = PC_LEFT;
+  snes->npcside = PC_LEFT;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
diff --git a/src/passo/inexact_newton.C b/src/passo/inexact_newton.C
index 6f2dbfb..009b840 100644
--- a/src/passo/inexact_newton.C
+++ b/src/passo/inexact_newton.C
@@ -209,7 +209,7 @@ SNESSolve_inexact_newton(SNES snes)
     if (flg == 1)
     {
       PCMGSetLevels(pc, levels, NULL);
-      PCMGSetGalerkin(pc, PETSC_TRUE);
+      PCMGSetGalerkin(pc, PC_MG_GALERKIN_PMAT);
 
       for (int i = 0; i < levels - 1; i++)
       {
@@ -223,9 +223,7 @@ SNESSolve_inexact_newton(SNES snes)
         Mat tmp1 = snes_passo->interpolations[i];
         PCMGSetInterpolation(pc, snes_passo->interpolations.size() - i, tmp1);
       }
-
-      PCMGSetNumberSmoothUp(pc, 3);
-      PCMGSetNumberSmoothDown(pc, 3);
+      PCMGSetNumberSmooth(pc, 3);
     }
 
     PASSO_CONVERGENCE_LOG ut_log;
@@ -290,7 +288,7 @@ SNESCreate_inexact_newton(SNES snes)
   snes->ops->reset = SNESReset_inexact_newton;
 
   snes->usesksp = PETSC_TRUE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
diff --git a/src/passo/linesearch.C b/src/passo/linesearch.C
index e40f413..b82fd9d 100644
--- a/src/passo/linesearch.C
+++ b/src/passo/linesearch.C
@@ -189,9 +189,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_Line_Search_Solver(SNES snes)
     snes->ops->reset          = SNESReset_passo_Line_Search_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
-    //snes->pcside = PC_LEFT;
+    //snes->npcside = PC_LEFT;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/multigrid.C b/src/passo/multigrid.C
index 16b6be3..78de75d 100644
--- a/src/passo/multigrid.C
+++ b/src/passo/multigrid.C
@@ -281,9 +281,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_multigrid(SNES snes)
     snes->ops->reset          = SNESReset_passo_multigrid;
 
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
 
-    //snes->pcside = PC_LEFT;
+    //snes->npcside = PC_LEFT;
 
     ierr       = PetscNewLog(snes,&neP); CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/newton.C b/src/passo/newton.C
index e3d18f0..eca532d 100644
--- a/src/passo/newton.C
+++ b/src/passo/newton.C
@@ -238,7 +238,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_Newton_Solver(SNES snes)
     snes->ops->reset          = SNESReset_passo_Newton_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/nml.C b/src/passo/nml.C
index 9a5bd2a..4f472f5 100644
--- a/src/passo/nml.C
+++ b/src/passo/nml.C
@@ -283,7 +283,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_nml(SNES snes)
     snes->ops->reset          = SNESReset_passo_nml;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/nonlinear_multigrid.C b/src/passo/nonlinear_multigrid.C
index 4acfb3e..d150560 100644
--- a/src/passo/nonlinear_multigrid.C
+++ b/src/passo/nonlinear_multigrid.C
@@ -252,7 +252,7 @@ SNESCreate_passo_NonlinearMultigrid_Solver(SNES snes)
   snes->ops->reset = SNESReset_passo_NonlinearMultigrid_Solver;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
diff --git a/src/passo/nullsolver.C b/src/passo/nullsolver.C
index 3428fe4..2a284e7 100644
--- a/src/passo/nullsolver.C
+++ b/src/passo/nullsolver.C
@@ -150,7 +150,7 @@ SNESCreate_passo_nullsolver(SNES snes)
   snes->ops->reset = SNESReset_passo_nullsolver;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
   // snes->pcside = PC_LEFT;
 
diff --git a/src/passo/rawnbgs.C b/src/passo/rawnbgs.C
index 8a40eed..bc1e6c9 100644
--- a/src/passo/rawnbgs.C
+++ b/src/passo/rawnbgs.C
@@ -66,7 +66,7 @@ PetscErrorCode SNESDestroy_passo_rawnbgs(SNES snes)
 PetscErrorCode SNESSetUp_passo_rawnbgs(SNES snes)
 {
     PetscFunctionBegin;
-    if (snes->pcside == PC_RIGHT) {SETERRQ(PETSC_COMM_SELF,PETSC_ERR_ARG_WRONGSTATE,"passo_rawnbgs only supports left preconditioning");}
+    if (snes->npcside == PC_RIGHT) {SETERRQ(PETSC_COMM_SELF,PETSC_ERR_ARG_WRONGSTATE,"passo_rawnbgs only supports left preconditioning");}
     if (snes->functype == SNES_FUNCTION_DEFAULT) snes->functype = SNES_FUNCTION_UNPRECONDITIONED;
     PetscFunctionReturn(0);
 }
@@ -143,7 +143,7 @@ SNESSolve_passo_rawnbgs(SNES snes)
     PetscPrintf(PETSC_COMM_WORLD, "SNESSolve_passo_rawnbgs #1,");
     RAWNBGS_CTX * rawnbgs_ctx = (RAWNBGS_CTX*)snes->data;
     
-    ierr = PetscCitationsRegister(SNESCitation,&SNEScite);CHKERRQ(ierr);
+//    ierr = PetscCitationsRegister(SNESCitation,&SNEScite);CHKERRQ(ierr);
     X = snes->vec_sol;
     F = snes->vec_func;
     B = snes->vec_rhs;
@@ -297,9 +297,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_rawnbgs(SNES snes)
     snes->ops->reset          = SNESReset_passo_rawnbgs;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
-    snes->pcside = PC_LEFT;
+    snes->npcside = PC_LEFT;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/richardson.C b/src/passo/richardson.C
index b5b79f7..fb88343 100644
--- a/src/passo/richardson.C
+++ b/src/passo/richardson.C
@@ -51,7 +51,7 @@ PetscErrorCode
 SNESSetUp_Petscpasso(SNES snes)
 {
   PetscFunctionBegin;
-  if (snes->pcside == PC_RIGHT)
+  if (snes->npcside == PC_RIGHT)
   {
     SETERRQ(
         PETSC_COMM_SELF, PETSC_ERR_ARG_WRONGSTATE, "Petscpasso only supports left preconditioning");
@@ -162,11 +162,11 @@ SNESSolve_Petscpasso(SNES snes)
   ierr = PetscObjectSAWsGrantAccess((PetscObject)snes);
   CHKERRQ(ierr);
 
-  if (snes->pc && snes->functype == SNES_FUNCTION_PRECONDITIONED)
+  if (snes->npc && snes->functype == SNES_FUNCTION_PRECONDITIONED)
   {
     ierr = SNESApplyNPC(snes, X, NULL, F);
     CHKERRQ(ierr);
-    ierr = SNESGetConvergedReason(snes->pc, &reason);
+    ierr = SNESGetConvergedReason(snes->npc, &reason);
     CHKERRQ(ierr);
     if (reason < 0 && reason != SNES_DIVERGED_MAX_IT)
     {
@@ -190,11 +190,11 @@ SNESSolve_Petscpasso(SNES snes)
     CHKERRQ(ierr);
     SNESCheckFunctionNorm(snes, fnorm);
   }
-  if (snes->pc && snes->functype == SNES_FUNCTION_UNPRECONDITIONED)
+  if (snes->npc && snes->functype == SNES_FUNCTION_UNPRECONDITIONED)
   {
     ierr = SNESApplyNPC(snes, X, F, Y);
     CHKERRQ(ierr); // put solution back
-    ierr = SNESGetConvergedReason(snes->pc, &reason);
+    ierr = SNESGetConvergedReason(snes->npc, &reason);
     CHKERRQ(ierr);
     if (reason < 0 && reason != SNES_DIVERGED_MAX_IT)
     {
@@ -292,7 +292,7 @@ SNESSolve_Petscpasso(SNES snes)
       CHKERRQ(ierr);
     }
 
-    if (snes->pc)
+    if (snes->npc)
     {
       if (snes->functype == SNES_FUNCTION_PRECONDITIONED)
       {
@@ -308,7 +308,7 @@ SNESSolve_Petscpasso(SNES snes)
         ierr = SNESApplyNPC(snes, X, F, Y);
         CHKERRQ(ierr);
       }
-      ierr = SNESGetConvergedReason(snes->pc, &reason);
+      ierr = SNESGetConvergedReason(snes->npc, &reason);
       CHKERRQ(ierr);
       if (reason < 0 && reason != SNES_DIVERGED_MAX_IT)
       {
@@ -352,9 +352,9 @@ SNESCreate_Petscpasso(SNES snes)
   snes->ops->reset = SNESReset_Petscpasso;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
-  snes->pcside = PC_LEFT;
+  snes->npcside = PC_LEFT;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
@@ -367,4 +367,4 @@ SNESCreate_Petscpasso(SNES snes)
     snes->stol = 1e-20;
   }
   PetscFunctionReturn(0);
-}
\ No newline at end of file
+}
diff --git a/src/passo/trustregion.C b/src/passo/trustregion.C
index edd2d80..33f5640 100644
--- a/src/passo/trustregion.C
+++ b/src/passo/trustregion.C
@@ -207,9 +207,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_Trust_region_Solver(SNES snes)
     snes->ops->reset          = SNESReset_passo_Trust_region_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
-    snes->pcside = PC_LEFT;
+    snes->npcside = PC_LEFT;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/trustregion_bound_contrains.C b/src/passo/trustregion_bound_contrains.C
index ad6479a..54eeb15 100644
--- a/src/passo/trustregion_bound_contrains.C
+++ b/src/passo/trustregion_bound_contrains.C
@@ -195,9 +195,9 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_Trust_region_bound_Solver(SNES snes
     snes->ops->reset          = SNESReset_passo_Trust_region_bound_Solver;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
-    snes->pcside = PC_LEFT;
+    snes->npcside = PC_LEFT;
     
     ierr       = PetscNewLog(snes,&neP);CHKERRQ(ierr);
     snes->data = (void*) neP;
diff --git a/src/passo/trustregion_petsc.C b/src/passo/trustregion_petsc.C
index 79ab9f3..c18f199 100644
--- a/src/passo/trustregion_petsc.C
+++ b/src/passo/trustregion_petsc.C
@@ -399,11 +399,11 @@ SNESSolve_passo_Trust_region_petsc_Solver(SNES snes)
     if (subproblem_type == SUBPROBLEM_STCG)
     {
       // std::cout<<"STCG ........ \n";
-      KSPSetType(snes->ksp, KSPSTCG);
-      KSPSTCGSetRadius(snes->ksp, delta);
+      KSPSetType(snes->ksp, KSPCGSTCG);
+      KSPCGSetRadius(snes->ksp, delta);
       KSPSolve(snes->ksp, G, S);
-      KSPSTCGGetObjFcn(snes->ksp, &pred);
-      KSPSTCGGetNormD(snes->ksp, &s_norm);
+      KSPCGGetObjFcn(snes->ksp, &pred);
+      KSPCGGetNormD(snes->ksp, &s_norm);
       KSPGetIterationNumber(snes->ksp, &lits);
       snes->linear_its += lits;
 
@@ -413,11 +413,11 @@ SNESSolve_passo_Trust_region_petsc_Solver(SNES snes)
     else if (subproblem_type == SUBPROBLEM_NASH)
     {
       // std::cout<<"NASH ........ \n";
-      KSPSetType(snes->ksp, KSPNASH);
-      KSPNASHSetRadius(snes->ksp, delta);
+      KSPSetType(snes->ksp, KSPCGNASH);
+      KSPCGSetRadius(snes->ksp, delta);
       KSPSolve(snes->ksp, G, S);
-      KSPNASHGetObjFcn(snes->ksp, &pred);
-      KSPNASHGetNormD(snes->ksp, &s_norm);
+      KSPCGGetObjFcn(snes->ksp, &pred);
+      KSPCGGetNormD(snes->ksp, &s_norm);
       KSPGetIterationNumber(snes->ksp, &lits);
       snes->linear_its += lits;
 
@@ -427,11 +427,11 @@ SNESSolve_passo_Trust_region_petsc_Solver(SNES snes)
     else if (subproblem_type == SUBPROBLEM_GLTR)
     {
       // std::cout<<"GLTR ........ \n";
-      KSPSetType(snes->ksp, KSPGLTR);
-      KSPGLTRSetRadius(snes->ksp, delta);
+      KSPSetType(snes->ksp, KSPCGGLTR);
+      KSPCGSetRadius(snes->ksp, delta);
       KSPSolve(snes->ksp, G, S);
-      KSPGLTRGetObjFcn(snes->ksp, &pred);
-      KSPGLTRGetNormD(snes->ksp, &s_norm);
+      KSPCGGetObjFcn(snes->ksp, &pred);
+      KSPCGGetNormD(snes->ksp, &s_norm);
       KSPGetIterationNumber(snes->ksp, &lits);
       snes->linear_its += lits;
 
@@ -571,9 +571,9 @@ SNESCreate_passo_Trust_region_petsc_Solver(SNES snes)
   snes->ops->reset = SNESReset_passo_Trust_region_petsc_Solver;
 
   snes->usesksp = PETSC_FALSE;
-  snes->usespc = PETSC_TRUE;
+  snes->usesnpc = PETSC_TRUE;
 
-  snes->pcside = PC_LEFT;
+  snes->npcside = PC_LEFT;
 
   ierr = PetscNewLog(snes, &neP);
   CHKERRQ(ierr);
diff --git a/src/passo/two_body.C b/src/passo/two_body.C
index 364319d..5d91b93 100644
--- a/src/passo/two_body.C
+++ b/src/passo/two_body.C
@@ -353,7 +353,7 @@ PetscErrorCode SNESSolve_passo_two_body(SNES snes)
         KSP contact_ksp;
         PC contact_pc;
         PCType type;
-        const MatSolverPackage stype;
+        MatSolverPackage stype;
         
         ierr = SNESGetKSP(snes_contact, &contact_ksp); CHKERRQ(ierr);
         ierr = KSPGetPC(contact_ksp, &contact_pc); CHKERRQ(ierr);
@@ -508,7 +508,7 @@ PETSC_EXTERN PetscErrorCode SNESCreate_passo_two_body(SNES snes)
     snes->ops->reset          = SNESReset_passo_two_body;
     
     snes->usesksp = PETSC_FALSE;
-    snes->usespc  = PETSC_TRUE;
+    snes->usesnpc  = PETSC_TRUE;
     
     ierr       = PetscNewLog(snes,&passo_ctx); CHKERRQ(ierr);
     snes->data = (void*) passo_ctx;
