diff --git a/framework/src/utils/PetscSupport.C b/framework/src/utils/PetscSupport.C
index 929ba1c027..611ef36705 100644
--- a/framework/src/utils/PetscSupport.C
+++ b/framework/src/utils/PetscSupport.C
@@ -354,7 +354,7 @@ petscConverged(KSP ksp, PetscInt n, PetscReal rnorm, KSPConvergedReason * reason
       break;
 #if !PETSC_VERSION_LESS_THAN(3, 6, 0) // A new convergence enum in PETSc 3.6
     case MOOSE_DIVERGED_PCSETUP_FAILED:
-      *reason = KSP_DIVERGED_PCSETUP_FAILED;
+      *reason = KSP_DIVERGED_PC_FAILED;
       break;
 #endif
     default:
