diff --git a/config/BuildSystem/config/packages/cuda.py b/config/BuildSystem/config/packages/cuda.py
index 55f5198a26..4f82265054 100644
--- a/config/BuildSystem/config/packages/cuda.py
+++ b/config/BuildSystem/config/packages/cuda.py
@@ -84,7 +84,8 @@ class Configure(config.package.Package):
 #    if self.getDefaultScalarType() == 'complex':
 #      raise RuntimeError('Must use real numbers with CUDA')
     if not config.setCompilers.Configure.isGNU(self.setCompilers.CC, self.log):
-      raise RuntimeError('Must use GNU compilers with CUDA')
+      #~ raise RuntimeError('Must use GNU compilers with CUDA')
+      pass
     if not self.getDefaultPrecision() in ['double', 'single']:
       raise RuntimeError('Must use either single or double precision with CUDA')
     else:
